bundle/jsky.app.ot/src/main/java/jsky/app/ot/gemini/editor/targetComponent/TelescopePosTableWidget.java

```diff
@@ -642,32 +666,31 @@ public void propertyChange(PropertyChangeEvent evt) {
             // Update the selection.
             if (rmTargets.isEmpty() && (addTargets.size() == 1)) {
                 // A single position was added, so just select it.
-                selectPos(addTargets.iterator().next());
+                selectTarget(addTargets.iterator().next());
                 return;
             }
 
             // If obs comp has a selected target, then honor it.
-            final SPTarget newSelectedTarget = TargetSelection.get(_env, _obsComp);
-            if (newSelectedTarget != null) {
-                final int index = _tableData.indexOf(newSelectedTarget);
-                _setSelectedRow(index);
+            final Option<SPTarget> newSelectedTarget = TargetSelection.getTargetForNode(_env, _obsComp);
+            if (newSelectedTarget.isDefined()) {
```

Comment:
That was exactly my thought. There are a lot of returns all over this method, which makes it difficult to use lambda expressions.